optuna/pruners/percentile.py

```diff
@@ -93,10 +99,18 @@ def prune(self, study, trial):
         if step is None:
             return False
 
-        if step <= self.n_warmup_steps:
+        n_warmup_steps = self.n_warmup_steps
+        if step <= n_warmup_steps:
             return False
 
-        if len(trial.intermediate_values) == 0:
+        lower_bound = (
+            (step - n_warmup_steps - 1) // self.interval_steps * self.interval_steps
+            + n_warmup_steps + 1)
+        assert lower_bound >= 0
```

Comment:
Maybe this is redundant since it just asserts `step >= 0` but I added it for code readability. I could remove it too.